docs/flow/components/discovery.transformer.md

```diff
@@ -0,0 +1,104 @@
+
+## Example
+
+```hcl
+discovery "transformer" "keep-backend-only" {
+  targets = [ 
+    { "__meta_foo" = "foo", "__address__" = "localhost", "instance" = "one",   "app" = "backend"  },
+    { "__meta_bar" = "bar", "__address__" = "localhost", "instance" = "two",   "app" = "database" },
+    { "__meta_baz" = "baz", "__address__" = "localhost", "instance" = "three", "app" = "frontend" }
+  ]
+  
+  relabel_config {
+    source_labels = ["__address__", "instance"]
+    separator     = "/"
+    target_label  = "destination"
+    action        = "replace"
+  } 
+  
+  relabel_config {
+    source_labels = ["app"]
+    action = "keep"
+    regex  = "backend"
+  }
+}
+```
+
+## Arguments
+
+The following arguments are supported and can be referenced by other
+components:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`targets` | `[]map[string]string` | The input targets | | no
+`relabel_config` | `RelabelConfig` | The relabeling steps to apply | | no
+
+
+### Targets
+The `targets` field contains the array of input targets. Each target consists
+of one or more string key-value pairs, referred to as its label set.
+
+### RelabelConfig
+The `relabel_config` block contains the definition of any relabeling rules
+that can be applied to an input target. If you want an overview of how
+relabeling works you can refer to the [Prometheus docs][] or this [blogpost][].
+
+#### Argument Reference
+The following arguments can be used to configure a `relabel_config` block.
+All arguments are optional and ny omitted fields will take on their default
+values.
+
+* `source_labels` - The list of labels whose values should be selected. Their content is concatenated using the `separator` and matched against `regex`.
+* `separator` - Used to concatenate the values selected from `source_labels`. Defaults to `;`.
+* `regex` - A valid RE2 expression with support for parenthesized capture groups. Used to match the extracted value from the combination of the `source_label` and `separator` fields or filter labels during the labelkeep/labeldrop/labelmap actions. Defaults to `(.*)`.
+* `modulus` - A positive integer used to calculate the modulus of the hashed source label values.
+* `target_label` - Label to which the resulting value will be written to.
+* `replacement` - The value against which a regex replace is performed, if the regex matched the extracted value. Supports previously captured groups. Defaults to `$1`.
+* `action` - The relabeling action to perform. Defaults to "replace".
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`output_targets` | `[]map[string]string` | The resulting targets' label set.
+
+The number of targets exposed in `output_targets` will be less or equal than
+the number of input targets, as some of them may be dropped by the
+relabeling process.
```

Comment:
It could be more too, if you use label mapping. I think we can just drop this paragraph.

Reply:
I'm okay with dropping the paragraph, but can they actually be more? I thought that labelmap will just rename labels from a specific target.

Reply:
It technically creates more. Given this label set: 

```
{ 
  "__label_foo" = "1", 
  "__label_bar" = "2",
}
```

And a relabel config of

```hcl
action      = "labelmap"
regex       = "__label_(.*)" 
replacement = "${1}" 
```

the resulting label set will be:

```
{ 
  "__label_foo" = "1", 
  "__label_bar" = "2", 
  "foo" = "1",
  "bar" = "2",
}
```

Reply:
Ahh, you're right, my bad :)